modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -1397,6 +1391,12 @@ private CPDefinition _updateNestedResources(
 			}
 		}
 
+		_cpDefinitionService.updateCPDefinitionAccountGroupFilter(
```

Comment:
this looks fine, but I think you should do the same fix for channel filter too

Reply:
or better add the reindex call to the rel services both of the channels and account group rels, there are lots of example in the code that do the reindex for the product.

Reply:
Hey @andrea-ale-sbarra, I thought about doing something like [this](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceChannelRelLocalServiceImpl.java#L349-L364), but the AccountGroupRelLocalService is a module outside of Commerce and can't depend on Commerce modules.
Also, the same fix for channel filter is not necessary since by the time it reaches this call, the channel filters will have already been updated and they will be included in the product document when being reindexed.